### PR TITLE
mariadb-el8

### PR DESCRIPTION
**Pull request by earthgecko**
Hi @fraenki 

I am working with your module and CentOS 8 using `galera::mariadb::default_version: '10.4'` I found that the galera package name under 10.4 is now galera-4, also the `libgalera_location` is now `/usr/lib64/galera-4/libgalera_smm.so` other than that it is running OK on CentOS with 10.4

I have seen that you have been updating this week and held off with a PR as I figured you may be making these updates.

I have also added a conditional check to adivse user that codership and percona vendors cannot run on el8 and as they have no el8 repos.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -405,6 +405,14 @@
   }
 
   if $configure_repo {
+    # Advise the user that codership and percona do not have a EL 8 repos
```
Comment (fraenki) [inline manifests/init.pp]: I'm undecided on this, these things get dated pretty quickly. I'd rather see this mentioned in the README under "Limitations".

**Comment by fraenki** [comment]
> I am working with your module and CentOS 8 using galera::mariadb::default_version: '10.4'

One note on this: You should rather use `$vendor_version` instead (as shown in the README), because `default_version` is an internal reference that could break without prior notice.

**Comment by earthgecko** [comment]
Changes as requested :+1: ,  Thanks for the advice on the `$vendor_version`.  I am actually declaring the `$vendor_version` variable in my yaml, I just changed the default locally to test that when that is changed to 10.4 all the other `galera::mariadb::104` variables behaved as expected, which they did.

**Comment by fraenki** [comment]
@earthgecko Merged, thanks!

**Pull request merged**